stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/plugin/listener/internal/ThreadQueryListenerFull.kt

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.offline.plugin.listener.internal
+
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.errors.ChatError
+import io.getstream.chat.android.client.extensions.internal.users
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.persistance.repository.MessageRepository
+import io.getstream.chat.android.client.persistance.repository.UserRepository
+import io.getstream.chat.android.client.plugin.listeners.ThreadQueryListener
+import io.getstream.chat.android.client.utils.Result
+import io.getstream.chat.android.client.utils.onSuccessSuspend
+import io.getstream.chat.android.offline.plugin.logic.channel.thread.internal.ThreadLogic
+import io.getstream.chat.android.offline.plugin.logic.internal.LogicRegistry
+import io.getstream.logging.StreamLog
+
+/**
+ * ThreadQueryListenerFull handles both state and database. It uses, if available, the database
+ * to update, if available, the state.
+ *
+ * @param logic [LogicRegistry] Optional class to handle state updates
+ * @param messageRepository [MessageRepository] Optional to handle database updates related to messages
+ * @param userRepository [UserRepository]  Optional to handle database updates related to user
+ * @param chatClient [ChatClient]
+ */
+internal class ThreadQueryListenerFull(
+    private val logic: LogicRegistry?,
+    private val messageRepository: MessageRepository?,
+    private val userRepository: UserRepository?,
+    private val chatClient: ChatClient,
```

Comment:
I wonder if we a pass just a suspending function here to avoid having unnecessary dependency:
```kotlin
getRemoteMessage: (messageId: String) -> Message
```